go-controller/pkg/ovn/gateway.go

```diff
@@ -1172,9 +1172,8 @@ func (gw *GatewayManager) syncNodeGateway(
 }
 
 func physNetName(netInfo util.NetInfo) string {
-	physnetName := types.PhysicalNetworkName
-	if netInfo.IsSecondary() {
-		physnetName = netInfo.GetNetworkName()
+	if netInfo.IsDefault() || netInfo.IsPrimaryNetwork() {
+		return types.PhysicalNetworkName
 	}
-	return physnetName
+	return netInfo.GetNetworkName()
```

Comment:
don't you need to adjust the L3 unit tests for this to pass ? 

I've adjusted it in https://github.com/ovn-org/ovn-kubernetes/pull/4561/commits/02dc5b23aaba350a3045865e43129554f50a0072 FWIW.